js/src/api/util/decode.js

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { isHex } from './types';
+
+import Func from '../../abi/spec/function';
+
+const CREATE_METHOD = '60606040';
+
+export function decodeCallData (data) {
+  if (!data || !data.length) {
+    return {};
+  }
+
+  if (!isHex(data)) {
+    throw new Error('Input to decodeCallData should be a hex value');
+  }
+
+  if (data.substr(0, 2) === '0x') {
+    return decodeCallData(data.slice(2));
+  } else if (data.length < 8) {
+    throw new Error('Input to decodeCallData should be method signature + data');
+  }
+
+  const signature = data.substr(0, 8);
+  const paramdata = data.substr(8);
+
+  if ((signature !== CREATE_METHOD) && paramdata.length % 64 !== 0) {
+    throw new Error(`Parameter length in decodeCallData not a multiple of 64 characters, ${paramdata.length}`);
+  }
+
+  return {
+    signature: `0x${signature}`,
+    paramdata: `0x${paramdata}`
+  };
+}
+
+export function decodeMethodInput (methodAbi, paramdata) {
+  if (!methodAbi) {
+    throw new Error('decodeMethodInput should receive valid method-specific ABI');
+  } else if (!paramdata) {
```

Comment:
Those checks don't have an `AND` relation, so it should be `else if` -> `if`. Every single failed check is a reason for `decodeMethodInput` to fail, but multiple can fail at once.

I that with `throw`, it doesn't make a big difference, but it's more robust.